### PR TITLE
Add refresh

### LINKED ISSUE #56: State of home page not getting updated automatically
**Describe the bug**
Home page isn't getting automatically updated after watching show/movie, and there is no option to manually refresh it except switch tabs.

**To Reproduce**
1. Watch something
2. Exit to home
3. Home page has outdated video position/old episode

**Expected behavior**
Home page is updated with new view position/next episode

**Device info (please complete the following information):**
 - Device: Poco X3 NFC
 - Android version: 11
 - App version: 0.2.1

### PR DESCRIPTION
**Pull request by lsrom**
- Adds ability to manually refresh home page
- Adds automatic refresh to onResume (switching tabs, coming back from player, orientation change)

Manual refresh shows default SwipeRefreshLayout indicator and NOT the usual linear one. This is because the indicator pops up anyway as user pulls down and if we hide it afterwards and show linear progress it looks bad imho. Also this way, user it indicates whether the refresh is automatic or manual.

EDIT: Second issue no longer matters, it is fixed. When the new content is same as old one, there should be no UI blinking.

Also, I did a bit of refactoring, sorry... :)

Closes #56

### CODE REVIEW

File: app/src/main/java/dev/jdtech/jellyfin/viewmodels/HomeViewModel.kt [hunk 1]
```diff
@@ -1,119 +1,112 @@
...
-                    if (!nextUpItems.isNullOrEmpty()) {
-                        items.add(HomeItem.Section(nextUpSection))
-                    }
+                    syncPlaybackProgress(repository, application)
                 }
+            } catch (e: Exception) {
+                Timber.e(e)
+                state.tryEmit(LoadingError(e.toString()))
+            }
+            state.tryEmit(Loading(inProgress = false))
```
Comment (jarnedemeulemeester) [inline app/src/main/java/dev/jdtech/jellyfin/viewmodels/HomeViewModel.kt:76]: The loading error is never shown because we emit Loading right after

**Review by jarnedemeulemeester — COMMENTED** [review]
The refresh works and the code looks pretty good. But maybe we should implement something like this for states: https://youtu.be/Qk2mIpE_riY?t=482 instead of only Loading and LoadingError
And also if we do we should create a new MR which changes all Fragments to use such states. We should discuss this on Discord

File: app/build.gradle.kts [hunk 2]
```diff
@@ -57,6 +57,7 @@ dependencies {
     implementation("androidx.core:core-ktx:1.6.0")
     implementation("androidx.core:core-splashscreen:1.0.0-alpha02")
     implementation("androidx.appcompat:appcompat:1.3.1")
+    implementation("androidx.swiperefreshlayout:swiperefreshlayout:1.2.0-alpha01")
```
[the remark this answers is not in the document]
Comment (lsrom) [reply app/build.gradle.kts:60]: Google's alpha libs are usually pretty stable I don't think there is an issue with using this but I can change it to older version if you want.

File: app/src/main/java/dev/jdtech/jellyfin/viewmodels/HomeViewModel.kt (hunk 1 above)
Comment (lsrom) [reply app/src/main/java/dev/jdtech/jellyfin/viewmodels/HomeViewModel.kt]: Good catch, thanks.

**Comment by lsrom** [comment]
I agree with the comment about states, didnt watch the entire video but I assume he handles the states similarly to MvRx (check it out if you are not familiar, it's pretty good). It would definitely be good to use something like that in the entire app as it would fix the LiveData mess currently in the code. Do you want me to add it to this MR?

File: app/build.gradle.kts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jarnedemeulemeester) [reply app/build.gradle.kts:62]: I would just stick to the stable version if the new version doesn't provide anything we can take advantage of. Just to be safe :)

**Comment by jarnedemeulemeester** [comment]
We will discuss the states situation later on Discord.

There still seems to be a few small issues. 

1. If I launch the app in airplane mode the loading progressbar stays on top and no error is show forever. 
2. If I launch the app with internet, disconnect when the home screen is loaded and refresh, the error is shown but the spinner of the refresh stays on screen. (Same issues with changing fragments but then it's the progressbar that stays)

**Pull request merged**